### PR TITLE
refactor(module): update module

### PR DESCRIPTION
**Pull request by ricardogobbosouza**
- Update module following `module-template`
- Coverage works again :smile: 
- Increase coverage (100%)
- Update `consola` to `^2.4.1` 
- ~~- Run module on hook `build:before`~~

Resolves #9 
Resolves #10 
Resolves #17 
Resolves #18 
Resolves #22

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<!--
+IMPORTANT: Please use the following link to create a new issue:
+
+https://nuxtjs.cmty.io/issues/new
```
Comment (TheAlexLichter) [inline .github/ISSUE_TEMPLATE.md]: I don't think we can use CMTY for that module at the moment

Comment (ricardogobbosouza) [reply .github/ISSUE_TEMPLATE.md]: @manniL 
I thought I would have a channel for each module
Should I remove this file?

Comment (ricardogobbosouza) [reply .github/ISSUE_TEMPLATE.md]: Removed!

File: README.md [hunk 2]
```diff
@@ -1,24 +1,33 @@
...
+[![Dependencies][david-dm-src]][david-dm-href]
+[![Standard JS][standard-js-src]][standard-js-href]
 
-[📖 **Release Notes**](./CHANGELOG.md)
-
-## Features
+> Separated environment variables for `server` and `client` build
 
-* Separated environment variables for `server` and `client` build
-* Thoroughly tested!
```
Comment (TheAlexLichter) [inline README.md:16]: Missing info here

File: lib/module.js [hunk 3]
```diff
@@ -1,45 +1,46 @@
...
-
+const consola = require('consola')
 const logger = consola.withScope('nuxt:separate-env')
 
-export default function separateEnv() {
-  const { server, client, ...env } = this.nuxt.options.env
-  // Set all other env variables except nested server/clients
-  this.nuxt.options.env = env
+function separateEnvModule() {
+  this.nuxt.hook('build:before', () => {
```
Comment (TheAlexLichter) [inline lib/module.js]: Why do you add the hook here? 🤔

**Review by TheAlexLichter — CHANGES_REQUESTED** [review]

File: lib/module.js (hunk 3 above)
Comment (ricardogobbosouza) [reply lib/module.js]: It really does not make sense

File: README.md (hunk 2 above)
Comment (ricardogobbosouza) [reply README.md:16]: This is more above, as a descriptive of the module
https://github.com/nuxt-community/separate-env-module/pull/19/files/13e31d7579a0b52ada2dc11383f641ca771fd985#diff-04c6e90faac2675aa89e2176d2eec7d8R10

**Comment by ricardogobbosouza** [comment]
@manniL Can merged?

File: README.md [hunk 4]
```diff
@@ -1,24 +1,33 @@
...
+[📖 **Release Notes**](./CHANGELOG.md)
 
 ## Setup
 
-- Add `@nuxtjs/separate-env` dependency to your project using yarn or npm
-- Add to `nuxt.config.js` modules - `modules: ["@nuxtjs/separate-env"]`
+1. Add `@nuxtjs/separate-env` dependency with `yarn` or `npm` into your project
+2. Add `@nuxtjs/separate-env` to `modules` section of `nuxt.config.js`:
```
Comment (TheAlexLichter) [inline README.md]:
```suggestion
2. Add `@nuxtjs/separate-env` to the `modules` section of `nuxt.config.js`:
```

File: README.md [hunk 5]
```diff
@@ -1,24 +1,33 @@
...
 
-* Separated environment variables for `server` and `client` build
-* Thoroughly tested!
+[📖 **Release Notes**](./CHANGELOG.md)
 
 ## Setup
 
-- Add `@nuxtjs/separate-env` dependency to your project using yarn or npm
-- Add to `nuxt.config.js` modules - `modules: ["@nuxtjs/separate-env"]`
+1. Add `@nuxtjs/separate-env` dependency with `yarn` or `npm` into your project
```
Comment (TheAlexLichter) [inline README.md]:
```suggestion
1. Add the `@nuxtjs/separate-env` dependency with `yarn` or `npm` to your project
```

File: lib/module.js [hunk 6]
```diff
@@ -52,4 +51,5 @@ function applyPrefix(definitionsToAdd) {
   }, {})
 }
 
+module.exports = separateEnvModule
```
Comment (TheAlexLichter) [inline lib/module.js:54]: Keep it ESM syntax please

File: package.json [hunk 7]
```diff
@@ -1,87 +1,56 @@
 {
   "name": "@nuxtjs/separate-env",
   "version": "0.0.1",
-  "description": "",
+  "description": "Separated environment variables for `server` and `client` build",
```
Comment (TheAlexLichter) [inline package.json]:
```suggestion
  "description": "Separated environment variables for server and client builds",
```

File: test/__snapshots__/module.test.js.snap [hunk 8]
```diff
@@ -78,29 +42,11 @@ exports[`ssr honor custom build.extend function 1`] = `
...
-  &quot;normal&quot;: &quot;Hi&quot;
-}</pre></div></div></div><script>window.__NUXT__=(function(a){return {layout:\\"default\\",data:[{type:a,env:{ONLY_CLIENT:\\"undefined\\",ONLY_SERVER:\\"yup\\",DIFFERENT:a,normal:\\"Hi\\"}}],error:null,serverRendered:true}}(\\"server\\"));</script><script src=\\"/_nuxt/runtime.js\\" defer></script><script src=\\"/_nuxt/pages/index.js\\" defer></script><script src=\\"/_nuxt/commons.app.js\\" defer></script><script src=\\"/_nuxt/vendors.app.js\\" defer></script><script src=\\"/_nuxt/app.js\\" defer></script>
+    <div data-server-rendered=\\"true\\" id=\\"__nuxt\\"><!----><div id=\\"__layout\\"><div id=\\"result\\"><h1>server</h1> <pre>[object Object]</pre></div></div></div><script>window.__NUXT__=(function(a){return {layout:\\"default\\",data:[{type:a,env:{ONLY_CLIENT:\\"undefined\\",ONLY_SERVER:\\"yup\\",DIFFERENT:a,normal:\\"Hi\\"}}],error:null,serverRendered:true}}(\\"server\\"));</script><script src=\\"/_nuxt/a2d7676f268f97a5849d.js\\" defer></script><script src=\\"/_nuxt/09889b896148317daca4.js\\" defer></script><script src=\\"/_nuxt/4da1fe42920b03139f11.js\\" defer></script><script src=\\"/_nuxt/feff3e7c475df22ce118.js\\" defer></script>
```
Comment (TheAlexLichter) [inline test/__snapshots__/module.test.js.snap]: There is definitely something wrong with the snapshots

**Review by TheAlexLichter — CHANGES_REQUESTED** [review]

File: lib/module.js (hunk 6 above)
Comment (ricardogobbosouza) [reply lib/module.js:54]: If it is not ESM we open coverage for Nuxt 1x
In my opinion we should release a version for Nuxt 1x and later a major release only covering Nuxt 2x
Conversation https://github.com/nuxt-community/redirect-module/pull/21

**Review by TheAlexLichter — APPROVED** [review]

**Pull request merged**